src/vdom/component.js

```diff
@@ -38,7 +38,7 @@ export function setComponentProps(component, props, opts, context, mountAll) {
 	component._disable = false;
 
 	if (opts!==NO_RENDER) {
-		if (opts===SYNC_RENDER || options.syncComponentUpdates!==false || !component.base) {
+		if ((opts&SYNC_RENDER) || options.syncComponentUpdates!==false || !component.base) {
```

Comment:
Is the change to `&` necessary? Not super familiar with all the incoming `opts`, but I do know that this is a deopt. Runtime engine has to infer that `1` or `0` are booleans, rather than just receive booleans directly. 

I think this might matter in performance given that this is the Component class & rendering is the default/assumed behavior.

Reply:
I've added a new `REATTEMPT_RENDER` opt. It behaves like `SYNC_RENDER`, but rethrows any errors that occur in child components. A reattempt path is required to cause any errors that occur in child components of an error handling path to bubble up to ancestor components instead of triggering an infinite rerender loop.

To clarify, imagine this scenario: Component renders a tree of child components. This child tree throws an error. Component catches the error and triggers a rerenders with a separate tree of child components describing the error. This tree also throws an error (yikes!). If we redispatched to the component's `componentDidCatch` it would likely trigger a rerender that generates the same erroring render tree. Instead preact should now treat that second error as having come from the component itself and pass it up to an ancestor so it can be handled at a higher level.

We can add a comparison with 0 or compare against both constants explicitly to remove the deopt path (at the expense of some additional bytes in either approach)